tests/Resources/Page/ODH/JupyterHub/JupyterLabLauncher.robot

```diff
@@ -235,11 +236,23 @@ Wait Until JupyterLab Is Loaded
   Wait Until Element Is Visible  xpath:${JL_TABBAR_CONTENT_XPATH}  timeout=${timeout}
 
 Clone Git Repository
+  [Documentation]    Clones git repository and logs error message if fails to clone
   [Arguments]  ${REPO_URL}
-  Navigate Home (Root folder) In JupyterLab Sidebar File Browser
-  Open With JupyterLab Menu  Git  Clone a Repository
-  Input Text  //div[.="Clone a repo"]/../div[contains(@class, "jp-Dialog-body")]//input  ${REPO_URL}
-  Click Element  xpath://div[.="CLONE"]
+  ${status}    ${err_msg} =    Run Keyword and Ignore Error    Clone Repo and Return Error Message    ${REPO_URL}
+    IF    "${status}" == "PASS"
+        ${dir_name} =    Get Directory Name From Git Repo URL    ${REPO_URL}
+        Delete Folder In User Notebook
+        ...    admin_username=${OCP_ADMIN_USER.USERNAME}
+        ...    username=${TEST_USER.USERNAME}
```

Comment:
@pranavsuryaw  This keyword will fail if the current user running the notebook is not ${TEST_USER.USERNAME}

To fix it I see two options:
- Add  ${admin_username} and ${username} as parameters to this keyword and the keywords using it, like `Clone Git Repository And Open` and `Clone Git Repository And Run`
- (I prefer this option Do not modify the existing clone keywords or add new ones and just add a new keyword to be explicitly called by the tests ). Something like this:
```
Delete Existing Git Clone Folder In User Notebook
    [Documentation]  In case it exists, deletes recursivelly existing git clone folder in user notebook
    ...   (e.g. /opt/app-root/src/repo-folder in ${username}'s notebook PVC.
    ...   Note: this command requires ${admin_username}  to be logged to the cluster (oc login ...)
    ...   and to have the user's notebook pod in Running state (e.g. jupyterhub-nb-ldap-2duser1)
    [Arguments]    ${admin_username}    ${username}    ${repo_url}

    ${repo_folder} =    Get Directory Name From Git Repo URL    ${repo_url}
    Delete Folder In User Notebook
    ...    admin_username=${admin_username}
    ...    username=${username}
    ...    folder=${repo_folder}

```

@lugi0 @bdattoma @pranavsuryaw What do you think?

Reply:
is the goal here to avoid failure in cloning when the same folder alredy exists in jupyterlab? 

If I got it right, I would say the @jgarciao 's solution of calling the kw to delete the folder from the test case may be good and cleaner. Maybe tht kw can be run with "Run Keyword and Ignore Error".

+1 on writing the error msg

Reply:
If I got it right, I would say the @jgarciao 's solution of calling the kw to delete the folder from the test case may still work by running the deletion kw with "Run Keyword and Ignore Error".

Otherwise, I think fetching the user from the URL is a good idea too. Regarding having a variable `CURRENT_USER`, I think it can bring confusion. I would rather prefer using the arguments

+1 on writing the error msg

Reply:
The agreement was to obtain the username using @lugi0 proposal